app/src/main/java/org/simple/clinic/util/ErrorResolver.kt

```diff
@@ -0,0 +1,43 @@
+package org.simple.clinic.util
+
+import io.reactivex.exceptions.CompositeException
+import okhttp3.internal.http2.ConnectionShutdownException
+import okhttp3.internal.http2.StreamResetException
+import java.net.SocketException
+import java.net.SocketTimeoutException
+import java.net.UnknownHostException
+
+object ErrorResolver {
+
+  fun resolve(error: Throwable): ResolvedError {
+    val actualCause = findActualCause(error)
+
+    return when (actualCause) {
+      is SocketException,
```

Comment:
I feel we can make this a set of Error types that can be checked against.

Reply:
That's a good idea, I'll do this.